tools/util/marshaller.go

```diff
@@ -0,0 +1,107 @@
+package util
+
+import (
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func MarshallObject(obj interface{}, writer io.Writer) error {
+	jsonBytes, err := json.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	var r unstructured.Unstructured
+	if err := json.Unmarshal(jsonBytes, &r.Object); err != nil {
+		return err
+	}
+
+	// remove status and metadata.creationTimestamp
+	unstructured.RemoveNestedField(r.Object, "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "spec", "template", "metadata", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "status")
+
+	/*	// remove dataSource from PVCs if empty
+		templates, exists, err := unstructured.NestedSlice(r.Object, "spec", "dataVolumeTemplates")
+		if exists {
+			for _, tmpl := range templates {
+				template := tmpl.(map[string]interface{})
+				_, exists, err = unstructured.NestedString(template, "spec", "pvc", "dataSource")
+				if !exists {
+					unstructured.RemoveNestedField(template, "spec", "pvc", "dataSource")
+				}
+			}
+			unstructured.SetNestedSlice(r.Object, templates, "spec", "dataVolumeTemplates")
+		}*/
```

Comment:
do we need to keep this commented block? If so, we might want to add a comment as to why it's been kept in.

Reply:
commented block removed, not required.